lib/kernel-manager.js

```diff
@@ -86,11 +86,26 @@ class KernelManager {
 
   startKernel(kernelSpec, grammar, onStarted) {
     const displayName = kernelSpec.display_name;
+    let kernelStartDir;
+    let currentPath = getEditorDirectory(store.editor);
 
     log("KernelManager: startKernel:", displayName);
 
+    switch (atom.config.get("Hydrogen.startDir")) {
+      case "firstProjectDir":
+        if (atom.project.getPaths().length > 0) {
+          kernelStartDir = atom.project.getPaths()[0];
+          break;
+        }
+      case "projectDirOfFile":
+        kernelStartDir = atom.project.relativizePath(currentPath)[0];
```

Comment:
Using `atom.project.relativizePath` can be a bit problematic since it will return `/` if no project is found or if there isn't a project related to the current file. I think it should fall back to `currentPath` instead of `/`. What do you think?